bazel/grpc_grpc.patch

```diff
@@ -0,0 +1,43 @@
+diff --git a/BUILD b/BUILD
+index e0827d46cb..5601c0214b 100644
+--- a/BUILD
++++ b/BUILD
+@@ -1612,6 +1612,9 @@ grpc_cc_library(
+         "src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_ev_driver.h",
+         "src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.h",
+     ],
++    copts = [
++        "-DCARES_STATICLIB",
```

Comment:
can we just pass this in windows/.bazelrc (i.e. `--copts=-DCARES_STATICLIB`) because other places use c-ares need this too. Then you don't need this patch file

Reply:
That wouldn't be optimal. Bazel is supposed to carry that water for us. In fact we tried various places to attach the;
```
    defines = ["CARES_STATICLIB]",
```
tag in grpc's BUILD, with no success. Perhaps @irengrig can comment?

Reply:
Oddly, this is already present in bazel/envoy_internals - as msvc_options returned by envoy_copts(). It's simply a problem that these foreign compilations such as grpc won't pick up our choices.

Reply:
envoy_copts isn't propagated to the dependencies so that is expected, so that's why I say add it in `.bazelrc`, perhaps you need it at toplevel.

Reply:
To clarify, I've tested and confirmed that the correct declaration of defines = ["TOKEN"] will propagate and we should be using this construct instead. I'll provide a more complete explanation and patch Monday.

Reply:
We pushed the fix to this PR branch. This drops the grpc_grpc.patch altogether, it isn't needed because we don't invoke grpc_cc_library, and it drops the individual *_STATICLIB defines, since the defines = ["token"] is carried onto all components depending on these CARES and NGHTTP2 static libs in our current bazel 0.29.1. Is ready to pull to master.

Reply:
We've escalated the defines=[] propagation issue in https://github.com/bazelbuild/rules_foreign_cc/issues/301 and agree with @lizan that this handful of -D'efines can live in .bazelrc for now. Adjusting the patch appropriately.

Reply:
> envoy_copts isn't propagated to the dependencies so that is expected, so that's why I say add it in `.bazelrc`, perhaps you need it at toplevel.

Just to be clear, every dependency of a dependency inherits its defines = [] list of -Dsymbol's, and this should be true even as two independent dependencies are intermediated by envoy's declaration of the dependency(-of-another-dependency). With a resolution to issue 301 we would be able to clean up the .bazelrc overrides.

Are there any outstanding questions, before we resolve this thread?